sql-plugin/src/main/31xdb/scala/com/nvidia/spark/rapids/shims/v2/Spark31XdbShimsBase.scala

```diff
@@ -129,5 +129,5 @@ trait Spark31XdbShimsBase extends SparkShims {
 
   override def shouldFallbackOnAnsiTimestamp(): Boolean = false
 
-  override def isCastingStringToNegDecimalScaleSupported: Boolean = true
+  override def isCastingStringToNegDecimalScaleSupported: Boolean = false
```

Comment:
this does seem a bit dangerous in the sense if we add a databricks 3.1.3 shim we could easily miss this.
perhaps at least put a comment here.  I assume the test won't fail if we get it wrong?

Reply:
The test isn't skipped for 3.1.3, it will blow up saying `part of the plan isn't columnar` as there is a check in `GpuCast` for it if we forget to override it for Spark-3.1.3-db. 

I can move this to Spark312dbShims.